charts/jenkins/templates/NOTES.txt

```diff
@@ -1,34 +1,43 @@
+{{- $prefix := .Values.controller.jenkinsUriPrefix | default "" -}}
+{{- $url := "" -}}
 1. Get your '{{ .Values.controller.adminUser }}' user password by running:
   kubectl exec --namespace {{ template "jenkins.namespace" . }} -it svc/{{ template "jenkins.fullname" . }} -c jenkins -- /bin/cat /run/secrets/additional/chart-admin-password && echo
-{{- if .Values.controller.ingress.hostName }}
-
-2. Visit http://{{ .Values.controller.ingress.hostName }}
+{{- if .Values.controller.ingress.hostName -}}
+{{- if .Values.controller.ingress.tls -}}
+{{- $url = print "https://" .Values.controller.ingress.hostName $prefix -}}
+{{- else -}}
+{{- $url = print "http://" .Values.controller.ingress.hostName $prefix -}}
+{{- end }}
+2. Visit {{ $url }}
 {{- else }}
 2. Get the Jenkins URL to visit by running these commands in the same shell:
-{{- if contains "NodePort" .Values.controller.serviceType }}
+{{- if contains "NodePort" .Values.controller.serviceType -}}
+{{- $url = print "http://$NODE_IP:$NODE_PORT" $prefix }}
```

Comment:
I think this won't work. NODE_IP and NODE_PORT are set after this line. Also not sure if you would have access to it.

Reply:
This is just a string that will be printed. These two variables aren't supposed to be expanded, they are just snippets for shell commands. This is how it was before as well.

Reply:
Yeah, that's true. If you'd like me to change something let me know.

Reply:
Ok, and I just realized that TLS were not being taken into account for some other cases as well. Pushing a fix.